pkg/cluster/k8sres.go

```diff
@@ -2228,6 +2241,9 @@ func (c *Cluster) generateConnectionPoolerPodTemplate(spec *acidv1.PostgresSpec)
 			ServiceAccountName:            c.OpConfig.PodServiceAccountName,
 			TerminationGracePeriodSeconds: &gracePeriod,
 			Containers:                    []v1.Container{poolerContainer},
+			Affinity: &v1.Affinity{
+				NodeAffinity: &spec.ConnectionPooler.NodeAffinity,
+			},
```

Comment:
Hm, should we also apply the configured `c.OpConfig.NodeReadinessLabel` here? Like we do in `generatePodTemplate` for db and backup pods.

Reply:
Makes sense. Fixed!